examples/flutter_gallery/lib/demo/customized/home.dart

```diff
@@ -0,0 +1,836 @@
+    );
+    _rotationAnimationController = AnimationController(
+      duration: const Duration(milliseconds: _kRotationAnimationDuration),
+      vsync: this,
+    );
+    _runnerAnimationController = AnimationController(
+      duration: const Duration(milliseconds: _kAnimateRunnerHeroFadeDuration),
+      vsync: this,
+    );
+    _numberCounterAnimationController = AnimationController(
+      duration: const Duration(milliseconds: _kAnimateNumberCounterDuration),
+      vsync: this,
+    );
+    _heroFadeInAnimation = _initAnimation(
+      from: 0.0,
+      to: 1.0,
+      curve: Curves.easeOut,
+      controller: _heroAnimationController,
+    );
+    _textFadeInAnimation = _initAnimation(
+      from: 0.0,
+      to: 1.0,
+      curve: Curves.easeIn,
+      controller: _textAnimationController,
+    );
+    _statsAnimationOne = _initAnimation(
+        from: 0.01,
+        to: 1.0,
+        curve: const Interval(0.2, 1.0, curve: Curves.easeOut),
+        controller: _statsAnimationControllerOne);
+    _statsAnimationTwo = _initAnimation(
+        from: 0.01,
+        to: 1.0,
+        curve: Curves.easeOut,
+        controller: _statsAnimationControllerTwo);
+    _statsAnimationThree = _initAnimation(
+        from: 0.01,
+        to: 1.0,
+        curve: Curves.easeOut,
+        controller: _statsAnimationControllerThree);
+    _statsAnimationFour = _initAnimation(
+        from: 0.01,
+        to: 1.0,
+        curve: Curves.easeOut,
+        controller: _statsAnimationControllerFour);
+    _heartAnimation = _initAnimation(
+        from: 1.0,
+        to: 1.25,
+        curve: Curves.easeOut,
+        controller: _heartAnimationController);
+    _rotationAnimation = _initAnimation(
+        from: 0.0,
+        to: 0.5,
+        curve: Curves.easeOut,
+        controller: _rotationAnimationController);
+    _runnerFadeAnimation = _initAnimation(
+        from: 0.0,
+        to: 1.0,
+        curve: Curves.easeOut,
+        controller: _runnerAnimationController);
+    _numberCounterAnimation = _numberCounterAnimationController;
+  }
+
+  void _configureThemes() {
+    _targetPlatform = Theme.of(context).platform;
+    _platformTextAlignment = _targetPlatform == TargetPlatform.android
+        ? TextAlign.left
+        : TextAlign.center;
+    _themeData = ThemeData(
+      primaryColor: const Color(0xFF212024),
+      buttonColor: const Color(0xFF3D3D3D),
+      iconTheme: const IconThemeData(color: Color(0xFF4A4A4A)),
+      brightness: Brightness.light,
+      platform: _targetPlatform,
+    );
+  }
+
+  Widget _contentWidget() {
+    final double screenHeight = MediaQuery.of(context).size.height;
+    final TargetPlatform platform = Theme.of(context).platform;
+    final String backTitle =
+        platform == TargetPlatform.android ? 'TrackFit' : '';
+    return Scaffold(
+      backgroundColor: const Color(0xFF212024),
+      body: NotificationListener<Notification>(
+        onNotification: _handleScrollNotification,
```

Comment:
I didn't read _handleScrollNotification yet but this means everything will rebuild every time you scroll. Ideally, let things that don't depend on the scroll position be in a different widget and don't trigger setState at the top level but use things that listen to listenables and rebuilds with the same subtree instance.

Reply:
The intention here is the fade out the content during the scroll event. I'll make sure we cache the elements that don't need to be rebuilt though.

Reply:
I removed the opacity and optimized the code a bit